lib/combinations.js

```diff
@@ -0,0 +1,18 @@
+function combinations(arr, size) {
+  var len = arr.length;
+
+  if (size > len) return [];
+  if (!size) return [[]];
+  if (size == len) return [arr];
+
+  return arr.reduce(function (acc, val, i) {
+    var res = combinations(arr.slice(i + 1), size - 1)
+    .map(function (comb) { return [val].concat(comb); });
+
+    return acc.concat(res);
```

Comment:
A couple of things in here. I don't think `res` needs to be a var - maybe let or const. 

I think the formatting needs some indentation, or a new variable on the second line of the reduce block. That could also use the es6 arrow syntax nicely as well `res.map((comb) => [val].concat(comb));`